src/main/java/io/github/genomicdatainfrastructure/daam/exceptions/ApplicationNotFoundException.java

```diff
@@ -0,0 +1,7 @@
+package io.github.genomicdatainfrastructure.daam.exceptions;
+
+public class ApplicationNotFoundException extends RuntimeException {
+    public ApplicationNotFoundException(String message) {
+        super(message);
```

Comment:
suggestion: to improve read,  the message is constant, you should move the text to this classe directly. And you should provide more meaningfull message, like the application id and the user id.